### PR TITLE
S3 detail page

### PR DESCRIPTION
**Pull request by matthewsearle01**

### CODE REVIEW

File: src/Models/DetailsModel.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Transformers\Models;
+use Transformers\Abstracts\Transformer;
+
+class DetailsModel
+{
+    public Transformer $transformerDetails;
+
+    public function __construct()
```
Comment (nealioa) [inline src/Models/DetailsModel.php]: This constructor is empty. Please remove it.

File: src/Models/DetailsModel.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Transformers\Models;
+use Transformers\Abstracts\Transformer;
+
+class DetailsModel
+{
+    public Transformer $transformerDetails;
```
Comment (nealioa) [inline src/Models/DetailsModel.php]: Either make this property private and add a getter, or keep public and delete the setter.
Talk to Henri.

File: src/Models/tests/DetailsModelTest.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        $result = $DetailsModel->setTransformer(null);
+    }
+    public function testSetTransformer_ThrowsException_GivenInvalidObject()
+    {
+        $DetailsModel = new DetailsModel();
+        $stub = $this->getMockBuilder(\stdclass::class)->getMock();
+        $this->expectException(TypeError::class);
+        $result = $DetailsModel->setTransformer($stub);
+    }
+    public function testSetTransformer_ThrowsException_GivenValidObject()
```
Comment (nealioa) [inline src/Models/tests/DetailsModelTest.php]: This test needs renaming as there is no exception thrown.

File: src/Models/tests/DetailsModelTest.php [hunk 4]
```diff
@@ -0,0 +1,56 @@
+<?php
+declare(strict_types=1);
+//namespace Transformers\Models\tests;
```
Comment (nealioa) [inline src/Models/tests/DetailsModelTest.php]: If the namespace isn't necessary, delete it.

File: tests/detailstest.php [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<?php
```
Comment (nealioa) [inline tests/detailstest.php]: Is this file needed?

**Review by nealioa — CHANGES_REQUESTED** [review]

File: src/Models/DetailsModel.php (hunk 1 above)
Comment (bnjn) [reply src/Models/DetailsModel.php]: Removed

File: src/Models/DetailsModel.php (hunk 2 above)
Comment (bnjn) [reply src/Models/DetailsModel.php]: Made private.

File: src/Models/tests/DetailsModelTest.php (hunk 3 above)
Comment (bnjn) [reply src/Models/tests/DetailsModelTest.php]: Renamed.

File: src/Models/tests/DetailsModelTest.php (hunk 4 above)
Comment (bnjn) [reply src/Models/tests/DetailsModelTest.php]: Removed.

File: tests/detailstest.php (hunk 5 above)
Comment (bnjn) [reply tests/detailstest.php]: File and folder deleted.

**Review by nealioa — DISMISSED** [review]

**Review by nealioa — APPROVED** [review]
